haystack/document_stores/weaviate.py

```diff
@@ -697,17 +697,31 @@ def _get_all_documents_in_index(
                 query = query.with_where(filter_dict)
 
             if all_docs:
-                # .with_limit() must be used with .with_offset, of the latter won't work properly
-                #   https://weaviate-python-client.readthedocs.io/en/latest/weaviate.gql.html?highlight=offset#weaviate.gql.get.GetBuilder.with_offset
+                # Passing offset:0 raises an error, so we pass it only after the first round
+                # `.with_limit()` must be used with `.with_offset`, or the latter won't work properly
+                # https://weaviate-python-client.readthedocs.io/en/latest/weaviate.gql.html?highlight=offset#weaviate.gql.get.GetBuilder.with_offset
                 query = query.with_limit(100).with_offset(offset=len(all_docs))
 
-            result = query.do()
+            try:
+                result = query.do()
+            except Exception as e:
+                raise ValueError(f"Weaviate raised an exception: {e}")
```

Comment:
We're trying to slowly convert the code to use custom exceptions. In `haystack.errors` there is a "base" `DocumentStoreError`: I think this is a great occasion to create `WeaviateDocumentStoreError(DocumentStoreError)` and use it. WDYT?